src/string-input/string-input.component.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * This file is part of ng2-json-editor.
+ * Copyright (C) 2017 CERN.
+ *
+ * ng2-json-editor is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * ng2-json-editor is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with ng2-json-editor; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+ * In applying this license, CERN does not
+ * waive the privileges and immunities granted to it by virtue of its status
+ * as an Intergovernmental Organization or submit itself to any jurisdiction.
+*/
+
+import {
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+  OnInit,
+  ChangeDetectionStrategy
+} from '@angular/core';
+
+import { DomUtilService, KatexService } from '../shared/services';
+
+@Component({
+  selector: 'string-input',
+  styleUrls: [
+    './string-input.component.scss'
+  ],
+  templateUrl: './string-input.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class StringInputComponent implements OnInit {
+
+  @Input() value: string;
+  @Input() path: Array<any>;
+  @Input() pathString: string;
```

Comment:
also not necessary if since `getElementById` could be avoided by using`@ViewChild` decorator. 


https://angular.io/docs/ts/latest/api/core/index/ViewChild-decorator.html